tests/integration/test_pipeline_processor_pool_router_config.py

```
@@ -0,0 +1,769 @@
+    def test_custom_rule_overrides_extension(self) -> None:
+        from file_organizer.pipeline.router import FileRouter, ProcessorType
+
+        router = FileRouter()
+        router.add_custom_rule(lambda p: p.name.startswith("vid_"), ProcessorType.VIDEO)
+        assert router.route(Path("vid_file.pdf")) == ProcessorType.VIDEO
+
+    def test_custom_rule_first_match_wins(self) -> None:
+        from file_organizer.pipeline.router import FileRouter, ProcessorType
+
+        router = FileRouter()
+        router.add_custom_rule(lambda p: True, ProcessorType.AUDIO)
```

Ruff (0.15.9)

[warning] Unused lambda argument: `p`

(ARG005)